planner/core/rule_aggregation_push_down.go

```diff
@@ -438,6 +438,16 @@ func (a *aggregationPushDownSolver) tryAggPushDownForUnion(union *LogicalUnionAl
 	if pushedAgg == nil {
 		return nil
 	}
+
+	// Update the agg mode for the pushed down aggregation.
```

Comment:
Could you explain why this update happens?

I don't get the logic to change 'CompleteMode' -> 'Partial1Mode'...

Reply:
**The behaviour between partition / final is different.**
Take `count(id)` for example:

```
a XX                  a 1                         a 1
b YY                  b 1                         b 2
c ZZ                  c 1                         c 2
     (partial mode)==>         (final mode) ==>  

b SS                  b 1
c KK                  c 1
```





In partial mode, the input is original data and turned into   "xx 1" where xx is `a` `b` `c`
In final mode, the input is partial data **and accumulated to "xx n"**

If you have this data as input:

```
a 1
b 2
c 2
```

In partial mode, the output is

```
a 1
b 1
c 1
```

In final mode, the output is

```
a 1
b 2
c 2
```


Back to the topic, we have push down agg to partition unions.
If the push-downed agg is not made to partial, 

```
create table t (c1 char, c2 int) partition by hash(c1) partitions 3;
insert into t values ('a', 1),('b', 2),('c', 3),('b', 4),('c', 5);
select count(*) from t group by c1;
```

We get wrong result:

```
a 1                   a 1                  a 1
b 2                   b 6                  b 6 
c 3  (final agg) =>   c 8  (final agg) =>  c 8
b 4
c 5
```

The correct result should be:

```
a 1                     a 1                  a 1
b 2                     b 2                  b 2
c 3  (partial agg) =>   c 2  (final agg) =>  c 2
b 4
c 5
```

Reply:
@bb7133